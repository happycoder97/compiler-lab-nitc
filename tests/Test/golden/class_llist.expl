type list {
    int data;
    list next;
}

class linkedlist {
    list head;
    list tail;
    int length;
    int getlength();
    int init();
    list insert(int data);
    int printlinkedlist();

    int getlength(){
        return self.length;
    }

    int init() {
        self.head=null;
        self.tail=null;
        self.length=0;
        return 1;
    }

    list insert(int data){
        list temp;
        temp=alloc();
        temp.data=data;
        temp.next=null;
        if (self.head== null) then
            self.head=temp;
            self.tail=temp;
        else
            self.tail.next=temp;
            self.tail=temp;
        endif;
        self.length=self.length+1;
        return temp;
    }

    int printlinkedlist(){
        list temp;
        temp=self.head;
        while (temp!= null) do
            write(temp.data);
        temp=temp.next;
        endwhile;
        return 1;
    }
}


linkedlist obj;

int main(){
    int x,y,z;
    list a;
    initialize();
    obj = new(linkedlist);
    x = obj.init();
    read(x);
    while(x!=0) do
        read(y);
        a = obj.insert(y);
        x = x-1;
    endwhile;
    write("Length");
    write(obj.getlength());
    write("List");
    x = obj.printlinkedlist();
    return 1;
}
