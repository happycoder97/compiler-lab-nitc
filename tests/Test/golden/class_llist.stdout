Length
6
List
2
9
4
3
2
0
